flink-runtime/src/test/java/org/apache/flink/runtime/io/disk/FileChannelManagerImplTest.java

```diff
@@ -0,0 +1,192 @@
+
+			deadline = Deadline.now().plus(TEST_TIMEOUT);
+			while (fileChannelManagerTestProcess.isAlive() && deadline.hasTimeLeft()) {
+				Thread.sleep(100);
+			}
+
+			if (fileChannelManagerTestProcess.isAlive()) {
+				fail("The file channel manager test process does not terminate in time, its output is: \n" + fileChannelManagerTestProcess.getProcessOutput());
+			}
+
+			// Checks if the directories are cleared.
+			assertThat("The file channel manager test process does not remove the tmp shuffle directories after termination, " +
+							"its output is \n" + fileChannelManagerTestProcess.getProcessOutput(),
+					fileOrDirExists(fileChannelDir, DIR_NAME_PREFIX),
+					is(false));
+		} finally {
+			if (fileChannelManagerTestProcess != null) {
+				fileChannelManagerTestProcess.destroy();
+			}
+		}
+	}
+
+	private boolean fileOrDirExists(File rootTmpDir, String namePattern) {
+		File[] candidates = rootTmpDir.listFiles((dir, name) -> name.contains(namePattern));
+		return candidates != null && candidates.length > 0;
+	}
+
+	/**
+	 * The {@link FileChannelManagerCleanupRunner} instance running in a separate JVM process.
+	 */
+	private static class FileChannelManagerTestProcess extends TestJvmProcess {
+		private final boolean callerHasHook;
+		private final String tmpDirectories;
+		private final String couldKillSignalFilePath;
+
+		public FileChannelManagerTestProcess(boolean callerHasHook, String tmpDirectories, String couldKillSignalFilePath) throws Exception {
+			this.callerHasHook = callerHasHook;
+			this.tmpDirectories = tmpDirectories;
+			this.couldKillSignalFilePath = couldKillSignalFilePath;
+		}
+
+		@Override
+		public String getName() {
+			return "File Channel Manager Test";
+		}
+
+		@Override
+		public String[] getJvmArgs() {
+			return new String[]{
+					Boolean.toString(callerHasHook),
+					tmpDirectories,
+					couldKillSignalFilePath
+			};
+		}
+
+		@Override
+		public String getEntryPointClassName() {
+			return FileChannelManagerCleanupRunner.class.getName();
+		}
+	}
+
+	/**
+	 * The entry point class to test the file channel manager cleanup with shutdown hook.
+	 */
+	public static class FileChannelManagerCleanupRunner {
+
+		public static void main(String[] args) throws Exception{
+			boolean callerHasHook = Boolean.parseBoolean(args[0]);
+			String tmpDirectory = args[1];
+			String couldKillSignalFilePath = args[2];
+
+			FileChannelManager manager = new FileChannelManagerImpl(new String[]{tmpDirectory}, DIR_NAME_PREFIX);
+
+			if (callerHasHook) {
+				ShutdownHookUtil.addShutdownHook(() -> manager.close(), "Caller", LOG);
+			}
+
+			// Single main process what we can be killed.
+			new File(couldKillSignalFilePath).createNewFile();
+
+			// Waits till get killed. If we have not killed in time, make sure we exit finally.
+			// Meanwhile, the test will fail due to process not terminated in time.
+			Thread.sleep(5 * TEST_TIMEOUT.toMillis());
```

Comment:
Can we still refer to the `TEST_TIMEOUT` for waiting before failure? Actually 50s is too long for this test I guess.

Reply:
I think the waiting test process would wait enough time before the `kill` process takes effect, otherwise we may have unstable scenarios. If the killing process cannot kill the process in `TEST_TIMEOUT`, then the test would fail directly with the timeout logic in the test function, and it will not waiting for `5 * TEST_TIMEOUT`.

Reply:
The normal case should take less than several seconds, so 30s should be enough for bearing the exception.